lib/Service/ItemService.php

```diff
@@ -315,7 +329,7 @@ public function starredCount($userId)
 
     /**
      * @param string $userId from which user the items should be taken
-     * @return array of items which are starred or unread
+     * @return int of items which are starred or unread
```

Comment:
This doesn't seem right either. The SQL query in this function definitely returns multiple rows if you have new feed items or starred ones. I tend to think that the test is not correct.

Reply:
You are probably right but how can the test work if something else than a int is returned
Maybe it returns int if it's only one and a array if it's more? :bomb:

Reply:
If I follow the code I would expect the custom type Entity to be returned
lib/public/AppFramework/Db/Mapper.php->findEntity which is deprecated since nc 14.0.0 returns a Entity
that is passed to findAllUnreadOrStarred in the ItemMapper and the service.

Didn't check what Entity is in it's core maybe php can also see it as an array, I don't have enough experience with php for that.

Reply:
I don't think so .. it returns an array of `Entity|Item`, not sure how this works with the test expecting an integer.
Any idea @SMillerDev?

Reply:
I think that's just a lazily written test. I didn't really get to those yet.